packages/plugin-core/src/features/windowDecorations.ts

```diff
@@ -114,6 +180,120 @@ export const debouncedUpdateDecorations = debounceAsyncUntilComplete({
   trailing: true,
 });
 
+export class NoteRefComment implements Comment {
```

Comment:
as per convention, can you move this to its own file - or at least the new code added here? This file is getting long.